docs/reference/sqrl/functions/custom-functions.md

```diff
@@ -14,3 +14,50 @@ Then follow the instructions below for implementing a function package in the la
 ## Functions in Java
 
 See the java user defined function example in our [examples](https://github.com/DataSQRL/datasqrl-examples/tree/main/user-defined-function).
+
+## Upload User-Defined Functions to the Repository
+
+If you want to upload your custom UDF to the Datasqrl repository, you need to take some extra steps.
+
+In the same folder where the JAR file resides (e.g., `myudf-0.1.0-SNAPSHOT.jar`), create the following files.
+
+### Create `package.json`
+
+The `package.json` file defines the package metadata, such as name, version, etc. A minimal `package.json` looks like this:
```

Comment:
We use the word "manifest" instead of "metadata" since the latter is overloaded.